Frontend/js/raytracing.js

```diff
@@ -21,86 +21,81 @@ function addShape() {
 		var size = parseInt(document.getElementById('size').value);
 		var color = document.getElementById('color').value;
 		
-		//alert message if not all values are correct
-		var modal = document.getElementById('myModal');
-		if( (x || x == 0) && (y || y == 0) && (z || z == 0) && (size || size == 0) ) {
-		}else{
-			document.getElementById("alertMessage").innerHTML = "Please fill all necessary Shape values(x, y, z, size).";
-			modal.style.display = "block";
-			return;
-			
-		}
+		//Validate and alert message if not all values are correct
 		
+		var validation_01 = true;

```

Comment:
Hi Farhad, do you think you can use a more descriptive variable name than 'validation_01' and 'validation_02'? At the first look, the variables don't convey the meaning that you might want to convey. Thanks!

Reply:
sure
will do it now.

Reply:
done

Reply:
Please let me know if any other changes are necessary at this point because I agree with Alex that we should have the loading and validation merged for tmr presentation. tnx